### DISCUSSION TITLE
Filament v2 : How to Trigger Action from Detail Modal with a Custom Button

### DISCUSSION THREAD

**Discussion by SupianIDz** [filamentphp/filament#9227] [Help] [answered]
Is there a way to trigger another table action via a custom button in the modal on the same data?

Example :
There are two action buttons in the table, let's say "detail" and "approve", normally before approving, the user will press the "detail" button first, then close the "detail" modal, and then press the "approval" button.

So, I want the user not to have to close the modal then press the approve button, I want to put the approve button on the detailed modal, if clicked, the "detail" modal will be closed and a confirmation modal of "approval" will automatically appear.

**Comment by leandrocfe** [comment]
Not sure but you could try extraModalActions
```php
->extraModalActions(fn () => [
    \Filament\Tables\Actions\Modal\Actions\Action::make('approval')
    ...
])
```

**Reply by SupianIDz** [reply]
Hi, that piece of code is almost what I want, but how do I get the button to work when clicked? like the action buttons on the table as usual?

In the image the approval button has a check mark icon, and the detail button has an eye icon.

<img width="515" alt="Screenshot 2023-10-22 at 04 00 04" src="https://github.com/filamentphp/filament/assets/37969970/357c87c4-2688-4f98-ae5a-f2452b6fdd4a">
<img width="635" alt="Screenshot 2023-10-22 at 03 59 16" src="https://github.com/filamentphp/filament/assets/37969970/f223a122-5080-41fd-bdaa-6f5e0a624af3">

**Reply by SupianIDz** [reply]
I tried to dive in, actions can't be in Closure form like other Action tables, they have to be strings, I tried passing the class name into 

```php
->action(App\Filament\Resources\ProductResource\Actions\ApprovalAction::class)
```

also doesn't work.

**Reply by leandrocfe** [reply, marked as answer]
try
```php
->extraModalActions(fn ($record) => [
    \Filament\Tables\Actions\Modal\Actions\Action::make('approval')
        ->action('approval', ['id' => $record])
])
```

ListPage or ManagePage

```php
public function approval($record): void
{
    //...
}
```

**Reply by SupianIDz** [reply]
Hi, it works fine, thank you. I'll look for a way to call the shorter `mountTableAction` 

This is the full version as I wanted.
```php
ViewAction::make()
->modalContent(function ($record) {
    return view('product.detail', compact('record'));
})
->extraModalActions(function (Product $record) {
    return [
        \Filament\Tables\Actions\Modal\Actions\Action::make('approval')->label('SETUJUI')->action('approval', [
            'id' => $record->id,
        ]),
    ];
}),

ApproveAction::make('approve'),
```

```php
public function approval($record)
{
    $this->mountTableAction('approve', json_decode($record)->id);
}
```
